### PR TITLE
Add Support for Root Admins to Submit Tax Forms

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/4805

Related Front-End PR: https://github.com/opencollective/opencollective-frontend/pull/8735

### CODE REVIEW

File: server/graphql/v2/mutation/RootMutations.ts [hunk 1]
```diff
@@ -251,4 +253,38 @@ export default {
       return moveExpenses(req, expenses, destinationAccount);
     },
   },
+  setTaxForm: {
+    type: new GraphQLObjectType({
+      name: 'SetTaxFormResult',
+      fields: {
+        success: { type: GraphQLBoolean },
```
Comment (Betree) [inline server/graphql/v2/mutation/RootMutations.ts:260]:
```suggestion
        success: { type: new GraphQLNonNull(GraphQLBoolean) },
```

File: server/graphql/v2/mutation/RootMutations.ts [hunk 2]
```diff
@@ -251,4 +253,38 @@ export default {
...
+      },
+    }),
+    description: '[Root only] A mutation to set the tax from for an account.',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Reference to the Account the tax form should be set.',
+      },
+      taxFormLink: {
+        type: new GraphQLNonNull(GraphQLNonEmptyString),
```
Comment (Betree) [inline server/graphql/v2/mutation/RootMutations.ts:270]: from `server/graphql/v2/scalar/URL.ts`

```suggestion
        type: new GraphQLNonNull(URL),
```

File: server/graphql/v2/mutation/RootMutations.ts [hunk 3]
```diff
@@ -251,4 +253,38 @@ export default {
...
+    async resolve(_, args, req) {
+      checkRemoteUserCanRoot(req);
+      // Always enforce 2FA for root actions
+      await twoFactorAuthLib.validateRequest(req, { requireTwoFactorAuthEnabled: true });
+
+      const account = await fetchAccountWithReference(args.account, { throwIfMissing: true });
+
+      await setTaxForm(account, args.taxFormLink, args.year);
+
+      return { success: true };
```
Comment (Betree) [inline server/graphql/v2/mutation/RootMutations.ts:287]: `setTaxForm` can return false if the document doesn't exist

```suggestion
      return { success: setTaxForm(account, args.taxFormLink, args.year) };
```

File: server/lib/tax-forms.ts [hunk 4]
```diff
@@ -111,6 +111,23 @@ const saveDocumentStatus = (account, year, requestStatus, data) => {
...
+    await legalDocument.update({
+      documentLink: taxFormLink,
+      year,
+      requestStatus: LEGAL_DOCUMENT_REQUEST_STATUS.RECEIVED,
+    });
+    return true;
+  } else {
+    return false;
+  }
+};
```
Comment (Betree) [inline server/lib/tax-forms.ts:129]: Would be clearer to me to just throw it if it doesn't exist

File: server/models/LegalDocument.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ export const LEGAL_DOCUMENT_TYPE = {
   US_TAX_FORM: 'US_TAX_FORM',
 };
 
-export const LEGAL_DOCUMENT_REQUEST_STATUS = {
+export const LEGAL_DOCUMENT_REQUEST_STATUS: Record<string, string> = {
```
[the remark this answers is not in the document]
Comment (SudharakaP) [reply server/models/LegalDocument.ts]: I thought we are being explicit when adding `Record<string, string>`. If we don't there will be a typescript error; 

```
server/lib/tax-forms.ts:123:7 - error TS2322: Type 'string' is not assignable to type 'Col | Fn | Literal | "ERROR" | "NOT_REQUESTED" | "REQUESTED" | "RECEIVED"'.

123       requestStatus: LEGAL_DOCUMENT_REQUEST_STATUS.RECEIVED,
```

**Review by Betree — APPROVED** [review]

**Pull request merged**